src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java

```diff
@@ -252,6 +233,44 @@ public OperationalResponse checkout(OperationalRequest request) {
 		resp.setConfirm(true);
 		return resp;
 	}
+
+	public ResponseEntity<?> totalFavesAndCheckouts(int userId) {
+		OperationalResponse confirm = new OperationalResponse();
+		if(userRepository.getUserCount(userId) > 0) {
+			int totalCheckouts = userRepository.getCheckoutCount(userId);
+			List<User> userFavorites = userRepository.getUserFavorites(userId);
+			List<Favorites> favorites = curateFavorites(userFavorites);
+			if(favorites == null) {
+				confirm.setMessage("Could not retrieve movie title");
+				return ResponseEntity.status(HttpStatus.NOT_FOUND).body(confirm);
+			}
+			FavesAndCheckOuts favesAndCheckOuts = new FavesAndCheckOuts();
+			favesAndCheckOuts.setCheckouts(totalCheckouts);
+			favesAndCheckOuts.setFavorites(favorites);
+			return ResponseEntity.status(HttpStatus.OK).body(favesAndCheckOuts);
+		}
+		confirm.setMessage( "User does not exist");
+		return ResponseEntity.status(HttpStatus.NOT_FOUND).body(confirm);
+	}
+
+	public List<Favorites> curateFavorites(List<User> userFavorites) {
+		System.out.println("::: " + userFavorites.size());
+		List<Favorites> favorites = new ArrayList<>();
+		for(User u : userFavorites) {
+			Favorites favorite = new Favorites();
+			SearchMoviesResponse searchAPIResp = APIClient.getAllMovies(new HashSet<>(Arrays.asList(u.getId().getProductId())));
```

Comment:
Why do we need to call search API multiple times? 

You can just loop all the users and create another set with all the movie ids and then call search with that list only once (to get movie info).

Please see implementation of getTopRated to see how to call search API with multiple movie ids.